src/valhalla_service.cc

```diff
@@ -12,29 +12,109 @@
 #include "baldr/rapidjson_utils.h"
 #include <boost/property_tree/ptree.hpp>
 
+#ifdef HAVE_HTTP
 #include <prime_server/http_protocol.hpp>
 #include <prime_server/prime_server.hpp>
 using namespace prime_server;
+#endif
 
 #include "midgard/logging.h"
 
 #include "loki/worker.h"
 #include "odin/worker.h"
 #include "thor/worker.h"
+#include "tyr/actor.h"
 
 int main(int argc, char** argv) {
-
-  if (argc < 2) {
+#ifdef HAVE_HTTP
+  if (argc < 2 || argc > 4) {
     LOG_ERROR("Usage: " + std::string(argv[0]) + " config/file.json [concurrency]");
+    LOG_ERROR("Usage: " + std::string(argv[0]) + " config/file.json action json_request");
+    return 1;
+  }
+#else
+  if (argc < 4) {
+    LOG_ERROR("Usage: " + std::string(argv[0]) + " config/file.json action json_request");
     return 1;
   }
+#endif
 
   // config file
   // TODO: validate the config
   std::string config_file(argv[1]);
   boost::property_tree::ptree config;
   rapidjson::read_json(config_file, config);
 
+  // one shot direct request mode
```

Comment:
this is the important part of this PR. if you want to make sure the only output is the response then you'll want to configure a `null` logger in your logging section of the config basically just set the logger type to empty string.

Reply:
alternatively we could force the null logger on in code here OR we could force the std::err logger so that only the real output goes to stdout? thoughts?

Reply:
I think it's fair expectation of a user to get the response only to `stdout`, so I'd force null logger or force logging to `stderr` only

Reply:
I agree. Most likely the user would like to use the response for further processing and the logger messages are a little in the way. Having only service response would be perfect. Setting `logging.type: ""` in the config doesn't help for [ANALYTICS] and a few other logs like 

```
2020/10/02 13:41:13.045647 [INFO] Tile extract successfully loaded with tile count: 7
2020/10/02 13:41:13.068255 [WARN] /data/valhalla/elevation/ currently has no elevation tiles
```

Reply:
updated the code to stderr logging so that only the important output lands on stdout